SMLHelper/Utility/EnumCacheManager.cs

```diff
@@ -27,10 +27,10 @@ internal class EnumCacheManager<T> where T : Enum
     {
         private class DoubleKeyDictionary : IEnumerable<KeyValuePair<int, string>>
         {
-            private readonly Dictionary<int, string> MapIntString = new Dictionary<int, string>();
-            private readonly Dictionary<T, string> MapEnumString = new Dictionary<T, string>();
-            private readonly Dictionary<string, T> MapStringEnum = new Dictionary<string, T>(StringComparer.InvariantCultureIgnoreCase);
-            private readonly Dictionary<string, int> MapStringInt = new Dictionary<string, int>(StringComparer.InvariantCultureIgnoreCase);
+            private readonly SortedDictionary<int, string> MapIntString = new SortedDictionary<int, string>();
+            private readonly SortedDictionary<T, string> MapEnumString = new SortedDictionary<T, string>();
+            private readonly SortedDictionary<string, T> MapStringEnum = new SortedDictionary<string, T>(StringComparer.InvariantCultureIgnoreCase);
+            private readonly SortedDictionary<string, int> MapStringInt = new SortedDictionary<string, int>(StringComparer.InvariantCultureIgnoreCase);
```

Comment:
What benefits does the `SortedDictionary` bring to this?  

Would the `SelfCheckingDictionary` have had something useful to offer here?

Reply:
The sorteddictionary makes it so that things in the cache are sorted by the number value of the techdata so things dont jump all over the place when removing and adding